### PR TITLE
SDAP-399: Quickstart update - Docker

### PR DESCRIPTION
**Pull request by RKuttruff**
Updated quickstart guide for standalone SDAP deployment with Docker.

~~Draft PR since this guide still needs to be verified and tested.~~

### CODE REVIEW

**Comment by RKuttruff** [comment]
Currently uses an older version of the nexus webapp image. Changes made to sdap after this image was made causes the `/timeSeriesSpark` endpoint to fail (This endpoint is the one used in the demo Jupyter notebook). I have implemented a patch for this issue in PR #186, pending approval, merging, and build + push of new image. If that happens, I will update the quickstart guide to use a more updated image.

**Comment by RKuttruff** [comment]
Currently investigating deployment with kubernetes. Will write second guide for that and add to this PR

**Comment by RKuttruff** [comment]
```
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
Input In [3], in <cell line: 22>()
     20 start_time = datetime(2015, 11, 1)
     21 end_time = datetime(2015, 11, 30)
---> 22 ts = nexuscli.time_series(datasets, bbox, start_time, end_time, spark=True)
     24 print("Time Series took {} seconds to generate".format(time.perf_counter() - start))
     26 # Plot the resulting time series using the helper method show_plot

File /opt/conda/lib/python3.10/site-packages/nexuscli-1.0-py3.10.egg/nexuscli/nexuscli.py:212, in time_series(datasets, bounding_box, start_datetime, end_datetime, spark)
    206     time_series_data = np.array([tuple(each.values()) for each in [entry for entry in data if entry['ds'] == i]])
    208     if len(time_series_data) > 0:
    209         time_series_result.append(
    210             TimeSeries(
    211                 dataset=response['meta'][i]['shortName'],
--> 212                 time=np.array([datetime.utcfromtimestamp(t).replace(tzinfo=UTC) for t in
    213                                time_series_data[:, key_to_index['time']]]),
    214                 mean=time_series_data[:, key_to_index['mean']],
    215                 standard_deviation=time_series_data[:, key_to_index['std']],
    216                 count=time_series_data[:, key_to_index['cnt']],
    217                 minimum=time_series_data[:, key_to_index['min']],
    218                 maximum=time_series_data[:, key_to_index['max']],
    219             )
    220         )
    222 return time_series_result

File /opt/conda/lib/python3.10/site-packages/nexuscli-1.0-py3.10.egg/nexuscli/nexuscli.py:212, in <listcomp>(.0)
    206     time_series_data = np.array([tuple(each.values()) for each in [entry for entry in data if entry['ds'] == i]])
    208     if len(time_series_data) > 0:
    209         time_series_result.append(
    210             TimeSeries(
    211                 dataset=response['meta'][i]['shortName'],
--> 212                 time=np.array([datetime.utcfromtimestamp(t).replace(tzinfo=UTC) for t in
    213                                time_series_data[:, key_to_index['time']]]),
    214                 mean=time_series_data[:, key_to_index['mean']],
    215                 standard_deviation=time_series_data[:, key_to_index['std']],
    216                 count=time_series_data[:, key_to_index['cnt']],
    217                 minimum=time_series_data[:, key_to_index['min']],
    218                 maximum=time_series_data[:, key_to_index['max']],
    219             )
    220         )
    222 return time_series_result

TypeError: 'numpy.str_' object cannot be interpreted as an integer```

**Comment by RKuttruff** [comment]
I instead plan to have the k8s deployment in another PR. Once I get this one sorted & validated, I will make it ready for review.

**Comment by RKuttruff** [comment]
Jupyter image verified to be working. 

Now awaiting validation & runtime observations from others.

PR will be ready when this happens.

**Comment by RKuttruff** [comment]
TODO: Check if AVHRR migration to Earthdata Cloud will affect granule download step.

File: docker/jupyter/Dockerfile
Comment (skorper) [inline docker/jupyter/Dockerfile:32]: This commit is from 2019... ideally, I think we'd like this to run on the latest code? What issues did you run when running this on master?

File: CHANGELOG.md
Comment (skorper) [inline CHANGELOG.md:47]: you can leave this section in empty in case we make a security change

File: docs/quickstart.rst
Comment (skorper) [inline docs/quickstart.rst:25]: 😱 why is so much disk space needed??

File: docs/quickstart.rst
Comment (skorper) [inline docs/quickstart.rst:202]: Perhaps some newlines here to make this more readable

```bash
docker run --name granule-ingester-1 --network sdap-net -e RABBITMQ_HOST="host.docker.internal:5672" \
	-e RABBITMQ_USERNAME="user" -e RABBITMQ_PASSWORD="bitnami" -d -e CASSANDRA_CONTACT_POINTS=host.docker.internal \
	-e CASSANDRA_USERNAME=cassandra -e CASSANDRA_PASSWORD=cassandra -e SOLR_HOST_AND_PORT="http://host.docker.internal:8983" \
	-v ${DATA_DIRECTORY}:/data/granules/ nexusjpl/granule-ingester:${GRANULE_INGESTER_VERSION}
```

File: docs/quickstart.rst
Comment (skorper) [inline docs/quickstart.rst:63]: let's update this back to nexusjpl before we merge this

File: docker/jupyter/requirements.txt
Comment (skorper) [inline docker/jupyter/requirements.txt:4]: I'm curious why we were able to remove this? Or why it was needed before but not now?

File: docker/jupyter/Dockerfile
Comment (RKuttruff) [reply docker/jupyter/Dockerfile:32]: I checked out this commit because I was unaware that the code for nexuscli had moved (I thought it was removed) so I grabbed a commit that would've been what was used to create the old docker image. I'll retry with the current version.

File: CHANGELOG.md
Comment (RKuttruff) [reply CHANGELOG.md:47]: I think I removed that line by accident. Oops

File: docs/quickstart.rst
Comment (RKuttruff) [reply docs/quickstart.rst:25]: Lot of docker images that are quite sizable.

+ a little margin for granule data &c so I just rounded up to a half GB or so

File: docs/quickstart.rst
Comment (RKuttruff) [reply docs/quickstart.rst:202]: Yea these commands get crazy long. I'll clean them up a bit.

File: docs/quickstart.rst
Comment (RKuttruff) [reply docs/quickstart.rst:83]: Will do.

File: docs/quickstart.rst
Comment (RKuttruff) [reply docs/quickstart.rst:63]: Ok. I'll let you know after I test the newer version of the image since I can't push to nexusjpl

File: docker/jupyter/requirements.txt
Comment (RKuttruff) [reply docker/jupyter/requirements.txt:4]: Not sure. The notebook runs just fine without it and it was causing build errors.

Comment (RKuttruff) [reply docker/jupyter/requirements.txt:4]: Possibly the notebook intended to do more. Or the the file was copy pasted from one that did and never checked.

File: CHANGELOG.md
Comment (RKuttruff) [reply CHANGELOG.md:47]: Well this is strange...
It's there in the branch

Comment (RKuttruff) [reply CHANGELOG.md:47]: https://github.com/RKuttruff/incubator-sdap-nexus/blob/quickstart-update/CHANGELOG.md?plain=1#L49

File: docs/quickstart.rst
Comment (RKuttruff) [reply docs/quickstart.rst:202]: Updated in this version: https://github.com/RKuttruff/incubator-sdap-nexus/blob/d3aac0187fe3bcf6008a12508bbf7b8078a0e4dd/docs/quickstart.rst

File: docs/quickstart.rst
Comment (RKuttruff) [reply docs/quickstart.rst:83]: Updated in this version: https://github.com/RKuttruff/incubator-sdap-nexus/blob/d3aac0187fe3bcf6008a12508bbf7b8078a0e4dd/docs/quickstart.rst

File: docker/jupyter/Dockerfile
Comment (RKuttruff) [reply docker/jupyter/Dockerfile]: Updated to point to nexuscli in master branch. Rebuilt & tested image
`rileykkjpl/jupyter:1.5`

File: CHANGELOG.md
Comment (RKuttruff) [reply CHANGELOG.md:47]: Added a newline after. It no longer shows as deleted

File: docs/quickstart.rst
Comment (ngachung) [inline docs/quickstart.rst:218]: We probably want to include this monitor.py with the quickstart rather than in @RKuttruff GitHub

Comment (RKuttruff) [reply docs/quickstart.rst:218]: I opened a PR to add the script to the repo. #205

File: docs/quickstart.rst
Comment (RKuttruff) [reply docs/quickstart.rst]: I shortened this by moving the environment variables into a file. Please let me know if that is an improvement.

File: docs/quickstart.rst
Comment (RKuttruff) [reply docs/quickstart.rst]: Headers updated

File: docs/quickstart.rst
Comment (RKuttruff) [reply docs/quickstart.rst:63]: I've built and tested a new version, `rileykkjpl/jupyter:1.5`. Once it's pushed to `nexusjpl` I can fix this line.

Comment (RKuttruff) [reply docs/quickstart.rst]: Updated

File: docs/quickstart.rst
Comment (RKuttruff) [reply docs/quickstart.rst]: Fixed

**Review by ngachung — APPROVED** [review]
Huge thank you @RKuttruff!

**Pull request merged**
